File: clase_python_01/MF0965_3_codigo_ejemplos/UF1290/Python/user_example.py
# UF1290 - Implementación e integración (Python)
class UserComponent:
    def __init__(self, username: str):
        self.username = username

class UserService:
    def register(self, user: UserComponent):
        print(f"User registered: {user.username}")

if __name__ == "__main__":
    user = UserComponent("alice")
    service = UserService()
    service.register(user)

File: clase_python_01/MF0965_3_codigo_ejemplos/UF1290/Java/UserExample.java

// UF1290 - Implementación e integración (Java)
public class UserComponent {
    private String username;

    public UserComponent(String username) {
        this.username = username;
    }

    public String getUsername() {
        return username;
    }
}

public class UserService {
    public void register(UserComponent user) {
        System.out.println("User registered: " + user.getUsername());
    }
}
